### DISCUSSION TITLE
v101.0.4951.53 not on Bromite F-Droid repo

### DISCUSSION THREAD

**Discussion by e00E** [bromite/bromite#2059] [General]
https://github.com/bromite/bromite/issues/2058
It seems like I'm not supposed to report this as a normal issue because the bot immediately closes it for not following the template and the template doesn't make sense for this issue.

Version on official fdroid repository: 100.0.4896.135
Version on Github : v101.0.4951.53
The Github version was released 2022-05-02 so at time of writing it has been 6 days without an update to fdroid.

This is similar to https://github.com/bromite/bromite/issues/2035 where it was commented that this would be fixed with the next releases. What is the cause for fdroid becoming out of sync with the repo?

**Comment by chaoscalm** [comment]
Just curiosity of mine, does F-Droid self-compile Bromite or it uses the pre-compiled builds available from github?

**Reply by csagan5** [reply]
> Just curiosity of mine, does F-Droid self-compile Bromite or it uses the pre-compiled builds available from github?

These are the same builds as available on GitHub, F-Droid is not involved at all as you will not find them on the F-Droid repository.

**Comment by rsssxd** [comment]
Bromite still hasn't received the latest security update.

**Reply by chaoscalm** [reply]
Unfortunate!

**Reply by csagan5** [reply]
> Bromite still hasn't received the latest security update.

Which latest security update?

**Reply by rsssxd** [reply]
https://www.forbes.com/sites/daveywinder/2022/05/10/new-google-chrome-security-warning-for-millions-of-smartphone-users/

Mulch browser has received the latest security update.

**Reply by rsssxd** [reply]
Updated Mulch (WebView) to Chromium 101.0.4951.61-1, has 13 security fixes.

https://divestos.org/index.php?page=news

**Reply by csagan5** [reply]
I was looking for a CVE; it is a good practice to mention CVEs when talking about security.

The article you mentioned points here: https://www.forbes.com/sites/daveywinder/2022/04/30/warning-massive-new-security-update-for-32-billion-google-chrome-users-confirmed/

Which talks about version 101.0.4951.41, which is older than the currently released Bromite.

> Updated Mulch (WebView) to Chromium 101.0.4951.61-1, has 13 security fixes.
> 
> https://divestos.org/index.php?page=news

Please post a list of CVEs that you think apply to Bromite and then anyone can verify.

**Reply by rsssxd** [reply]
https://github.com/GrapheneOS/Vanadium/commit/3f3b8f806e097421a54f3583fe41c392276e402e

**Reply by csagan5** [reply]
> Please post a list of CVEs that you think apply to Bromite and then anyone can verify.

**Reply by rsssxd** [reply]
😌😌

https://chromereleases.googleblog.com/2022/05/chrome-for-android-update.html?m=1

**Reply by rsssxd** [reply]
[$5000][1316990] High CVE-2022-1633: Use after free in Sharesheet. Reported by Khalil Zhani on 2022-04-18[$3000][1314908] High CVE-2022-1634: Use after free in Browser UI. Reported by Khalil Zhani on 2022-04-09[$3000][1319797] High CVE-2022-1635: Use after free in Permission Prompts. Reported by Anonymous on 2022-04-26[$NA][1297283] High CVE-2022-1636: Use after free in Performance APIs. Reported by Seth Brenith, Microsoft on 2022-02-15[$TBD][1311820] High CVE-2022-1637: Inappropriate implementation in Web Contents. Reported by Alesandro Ortiz on 2022-03-31[$TBD][1316946] High CVE-2022-1638: Heap buffer overflow in V8 Internationalization. Reported by DoHyun Lee (@l33d0hyun) of DNSLab, Korea University on 2022-04-17[$TBD][1317650] High CVE-2022-1639: Use after free in ANGLE. Reported by SeongHwan Park (SeHwa) on 2022-04-19[$TBD][1320592] High CVE-2022-1640: Use after free in Sharing. Reported by Weipeng Jiang (@Krace) and Guang Gong of 360 Vulnerability Research Institute on 2022-04-28[$5000][1305068] Medium CVE-2022-1641: Use after free in Web UI Diagnostics. Reported by Rong Jian of VRI on 2022-03-10

**Reply by uazo** [reply]
you might also be right, but I would remember that bromite specializes in privacy, not security in general.
there are other opensource browsers that have this as their goal
for each of those cve we should check if it is possible to exploit them in android to have an accurate evaluation.
I realize it would be better, but it's still a free time hobby.

**Reply by rsssxd** [reply]
Mulch and Vanadium released the security update the same day. This should never be difficult for Bromite.

The blog says the security update is for Android.

**Reply by csagan5** [reply]
Unfortunately all those issues are locked so details cannot be read about them and that prevents assessing whether they affect Bromite or not; for some of them it is most likely yes.

> Mulch and Vanadium released the security update the same day. This should never be difficult for Bromite.

Whether difficult or not there is no such guarantee or promise anywhere; if you have a problem with that then do not use it.

**Reply by rsssxd** [reply]
It's sad to hear that. If there is no security, there is no privacy.

I will switch to using Mulch again

**Comment by csagan5** [comment]
> Version on official fdroid repository: 100.0.4896.135

I cannot confirm, the Bromite official F-Droid repository contains the latest `101.0.4951.53` since its release.

**Reply by e00E** [reply]
This is correct as of now (you commented 3 hours ago). It was not correct when I created the issue (5 days ago). Are you saying the report was wrong 5 days ago already?
Checking in the f-droid app I do see that it reports that 101.0.4951.53 was added 2022-05-02 however I am very sure it did not show this version when I created the discussion because I made sure to update the repositories and double checked. Could this somehow be an issue with the Bromite repository server? Maybe it caches old state for too long? Does the "added on" line on Fdroid accurate reflect the date at which the package was uploaded (it could instead show some metadata field of the app that is set to this date, but the app itself might have been added to the repository later)?

**Reply by csagan5** [reply]
> This is correct as of now (you commented 3 hours ago). It was not correct when I created the issue (5 days ago). Are you saying the report was wrong 5 days ago already?

Yes, all index files were last updated at `2022-05-03 01:44 CET`, you can see it also when downloading the files via web browser.

> Could this somehow be an issue with the Bromite repository server? Maybe it caches old state for too long?

The server does not cache anything, clients may.

> Does the "added on" line on Fdroid accurate reflect the date at which the package was uploaded (it could instead show some metadata field of the app that is set to this date, but the app itself might have been added to the repository later)?

You would still see the entry in the F-Droid client but then download would fail, if that were the case.

**Reply by e00E** [reply]
Thank you for the clarifications. I'm still quite sure that I checked this correctly and that I did not see the version in Fdroid but I could imagine that this is a bug in Fdroid itself. Should this happen again I'll try to manually confirm what exactly the bromite fdroid repository contains compared to what the fdroid client shows me.

**Reply by csagan5** [reply]
Maybe it is something stored in the F-Droid client cache that did not get invalidated; it is possible that it is due to something on the F-Droid repository but can't think of anything right now.

Next time you can try erasing the F-Droid client cache and that should offer some clues. Perhaps the logs will also tell something.
